src/rewritable_units/element.rs

```diff
@@ -413,6 +419,37 @@ impl<'r, 't> Element<'r, 't> {
         }
     }
 
+    /// Replaces the element start tag.
+    ///
+    /// Consequent calls to the method overwrite a previously replaced start tag.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use lol_html::{rewrite_str, element, RewriteStrSettings};
+    /// use lol_html::html_content::ContentType;
+    ///
+    /// let html = rewrite_str(
+    ///     r#"<div id="foo">Hello</div>"#,
+    ///     RewriteStrSettings {
+    ///         element_content_handlers: vec![
+    ///             element!("#foo", |el| {
+    ///                 el.replace_start_tag("<div>");
+    ///
+    ///                 Ok(())
+    ///             })
+    ///         ],
+    ///         ..RewriteStrSettings::default()
+    ///     }
+    /// ).unwrap();
+    ///
+    /// assert_eq!(html, r#"<div>Hello</div>"#);
```

Comment:
can you please use the actual test case you care about (which I gather is something like `if el.is_self_closing() { el.replace_start_tag() }`) ?

Reply:
I replied in the other comment.  Basically my use case is not a common one and is hard for a reader to understand.  This is simple and to the point, which is what makes better documentation imo.